helm/provider.go

```diff
@@ -373,10 +375,55 @@ func (m *Meta) GetHelmConfiguration(namespace string) (*action.Configuration, er
 	if err := actionConfig.Init(kc, namespace, m.HelmDriver, debug); err != nil {
 		return nil, err
 	}
+
 	debug("[INFO] GetHelmConfiguration success")
 	return actionConfig, nil
 }
 
+// dataGetter lets us call Get on both schema.ResourceDiff and schema.ResourceData
+type dataGetter interface {
+	Get(key string) interface{}
+}
+
+// OCIRegistryLogin creates an OCI registry client and logs into the registry if needed
+func OCIRegistryLogin(actionConfig *action.Configuration, d dataGetter) error {
+	registryClient, err := registry.NewClient()
+	if err != nil {
+		return fmt.Errorf("could not create OCI registry client: %v", err)
+	}
+	actionConfig.RegistryClient = registryClient
+
+	// log in to the registry if neccessary
+	repository := d.Get("repository").(string)
+	chartName := d.Get("name").(string)
```

Comment:
Shall we use here `chart`(chart name) instead of `name`(release name)?

Reply:
Yep you're right, good catch